lib/ide-haskell-repl-view.coffee

```diff
@@ -196,6 +199,20 @@ class IdeHaskellReplView
   setPrompt: (prompt) ->
     @promptDiv.innerText = prompt + '>'
 
+  getCompletions: (prefix) =>
+    return [] if prefix == '' or prefix == ' '
+    @ghci.sendCompletionRequest prefix
+    new Promise (resolve) =>
+      @completionHandler = @ghci.emitter.on 'complete', (lines) =>
+        suggestions = []
+        for line, i in lines when i > 0
+          lines[i] = line.replace /^"|"$/g, ''
+        results = filter(lines, prefix, {maxResults: 32})
+        for result in results
+          suggestions.push {text: result}
+        resolve (suggestions)
+        if @completionHandler? then @completionHandler.dispose()
```

Comment:
`completionHandler?.dispose?()` would be more concise. Although if you switch to local variable, it should always be defined.